src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1937,35 +1937,44 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
                 return;
             }
 
-            // TODO: report more specific diagnostics here for failed method group conversions
-            if (operand.Kind == BoundKind.MethodGroup)
+            switch (operand.Kind)
             {
-                diagnostics.Add(ErrorCode.ERR_NoExplicitConv, syntax.Location, MessageID.IDS_SK_METHOD.Localize(), targetType);
-                return;
-            }
-
-            if (operand.Kind == BoundKind.TupleLiteral)
-            {
-                var tuple = (BoundTupleLiteral)operand;
-                var targetElementTypes = default(ImmutableArray<TypeSymbol>);
+                case BoundKind.MethodGroup:
+                    {
+                        // TODO: report more specific diagnostics here for failed method group conversions
```

Comment:
Oh, I see you've just moved a comment. Never mind.

---
In reply to: [133061608](https://github.com/dotnet/roslyn/pull/21006#discussion_r133061608) [](ancestors = 133061608)